### PR TITLE
[SYCLomatic] Support the migration for some 64-bit version cuBLAS API, refine some 32-bit version cuBLAS API and fix some warning info in the inc file

### PR DESCRIPTION
**Pull request by zhiweij1**
Support below 30 64-bit version API:
`cublas[S|D|C|Z]syrk_64`,
`cublas[C|Z]herk_64`,
`cublas[S|D|C|Z]syr2k_64`,
`cublas[C|Z]her2k_64`,
`cublas[S|D|C|Z]symm_64`,
`cublas[C|Z]hemm_64`,
`cublas[S|D|C|Z]trsm_64`,
`cublas[S|D|C|Z]geam_64`,
`cublas[S|D|C|Z]dgmm_64`

Refined below 22 32-bit version API:
`cublas[S|D|C|Z]syrk`,
`cublas[S|D|C|Z]symm`,
`cublas[C|Z]hemm`,
`cublas[S|D|C|Z]trsm`,
`cublas[S|D|C|Z]geam`,
`cublas[S|D|C|Z]dgmm`

Fix 4 API warning info in the inc file:
`cublas[S|D|C|Z]gemm_64`

Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/test/dpct/cublas-usm.cu [hunk 1]
```diff
@@ -740,10 +740,10 @@ void foo3() {
...
-  //CHECK:oneapi::mkl::blas::column_major::dgmm_batch(*handle, oneapi::mkl::side::left, m, n, a_f, lda, 0, x_f, incx, 0, c_f, ldc, ldc * n, 1);
-  //CHECK-NEXT:oneapi::mkl::blas::column_major::dgmm_batch(*handle, oneapi::mkl::side::left, m, n, a_d, lda, 0, x_d, incx, 0, c_d, ldc, ldc * n, 1);
-  //CHECK-NEXT:oneapi::mkl::blas::column_major::dgmm_batch(*handle, oneapi::mkl::side::left, m, n, (std::complex<float>*)a_f, lda, 0, (std::complex<float>*)x_c, incx, 0, (std::complex<float>*)c_c, ldc, ldc * n, 1);
-  //CHECK-NEXT:oneapi::mkl::blas::column_major::dgmm_batch(*handle, oneapi::mkl::side::left, m, n, (std::complex<double>*)a_z, lda, 0, (std::complex<double>*)x_z, incx, 0, (std::complex<double>*)c_z, ldc, ldc * n, 1);
+  //CHECK:oneapi::mkl::blas::column_major::dgmm(*handle, oneapi::mkl::side::left, m, n, a_f, lda, x_f, incx, c_f, ldc);
+  //CHECK-NEXT:oneapi::mkl::blas::column_major::dgmm(*handle, oneapi::mkl::side::left, m, n, a_d, lda, x_d, incx, c_d, ldc);
+  //CHECK-NEXT:oneapi::mkl::blas::column_major::dgmm(*handle, oneapi::mkl::side::left, m, n, (std::complex<float>*)(sycl::float2*)a_f, lda, (std::complex<float>*)x_c, incx, (std::complex<float>*)c_c, ldc);
```
Comment (zhimingwang36) [inline clang/test/dpct/cublas-usm.cu]: can  "(std::complex<float>*)(sycl::float2*)" be simplified?

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
Please update the commit message to include all code changes.

File: clang/test/dpct/cublas-usm.cu (hunk 1 above)
Comment (zhiweij1) [reply clang/test/dpct/cublas-usm.cu]: fixed.

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**